tests/integration/test_cubepart.py

```diff
@@ -1013,6 +1013,47 @@ def it_uses_column_proportions_for_pop_counts_when_column_dim_is_cat_date(self):
             abs=10e-2,
         )
 
+    def it_uses_correct_proportion_for_cat_array_subvariables(self):
+        slice_ = Cube(CR.SIMPLE_CAT_ARRAY, population=100).partitions[0]
+        assert slice_.population_counts == pytest.approx(
+            np.array(
+                [
+                    [50.0, 50.0, 0.0, 0.0],
+                    [16.66666667, 50.0, 33.33333333, 0.0],
+                    [0.0, 33.33333333, 16.66666667, 50.0],
+                ]
+            )
+        )
+        # ---Assert population is equal to total for each subvariable
+        assert slice_.population_counts.sum(axis=1) == pytest.approx(
+            np.array([100, 100, 100])
+        )
+
+    def it_uses_correct_proportion_for_pets_array(self):
```

Comment:
i would use the name of the test with the same structure of the fixture. So I would say:
`_for_mr_x_ca_cat_x_ca_subvar` and not `pets_array`